src/builder/QuarkBuilder.sol

```diff
@@ -262,37 +262,30 @@ contract QuarkBuilder {
 
     // Assert that each chain has sufficient funds to cover the max cost for that chain.
     // Check user account can cover the cost of each actions
-    function assertActionsAffordable(Actions.Action[] memory actions, Accounts.ChainAccounts[] memory chainAccountsList)
-        internal
-        pure
-    {
-        Actions.Action[] memory bridgeActions = findActionsOfType(actions, Actions.ACTION_TYPE_BRIDGE);
-        Actions.Action[] memory transferActions = findActionsOfType(actions, Actions.ACTION_TYPE_TRANSFER);
+    function assertActionsAffordable(
+        Actions.Action[] memory actions,
+        Accounts.ChainAccounts[] memory chainAccountsList,
+        TransferIntent memory transferIntent
+    ) internal pure {
+        Actions.Action[] memory bridgeActions = Actions.findActionsOfType(actions, Actions.ACTION_TYPE_BRIDGE);
+        Actions.Action[] memory transferActions = Actions.findActionsOfType(actions, Actions.ACTION_TYPE_TRANSFER);
 
         uint256 plannedBridgeAmount = 0;
-        // Verify bridge actions is affordable, and update plannedBridgeAmount for verifying transfer actions
+        // Verify bridge actions are affordable, and update plannedBridgeAmount for verifying transfer actions
         for (uint256 i = 0; i < bridgeActions.length; ++i) {
             Actions.BridgeActionContext memory bridgeActionContext =
                 abi.decode(bridgeActions[i].actionContext, (Actions.BridgeActionContext));
-
+            uint256 paymentAssetBalanceOnChain = Accounts.sumBalances(
+                Accounts.findAssetPositions(bridgeActionContext.token, bridgeActions[i].chainId, chainAccountsList)
```

Comment:
This has to be `bridgeActions[i].paymentToken`, not `bridgeActionContext.token` right? Would be good to have some tests because these bugs slipping through is not great :/

I suggest writing the test in a follow-up PR since this one is getting large and would be great to have it merged soon.

Reply:
yeah. as for now all our test suites are based on USDC. 
Fxied.